meta-balena-common/recipes-containers/resin-supervisor/resin-supervisor/start-resin-supervisor

```diff
@@ -76,7 +76,7 @@ runSupervisor() {
         --restart=always \
         --net=host \
         --cidenv=SUPERVISOR_CONTAINER_ID \
-        -v /var/run/balena-engine.sock:/var/run/balena-engine.sock \
+        -v /var/run:/var/run \
```

Comment:
Is the supervisor going to be OK with mounting the whole /var/run directory? It's possible there's stuff inside the supervisor container itself that creates sockets there and it might conflict with the host (dbus, avahi?). cc @CameronDiver @hedss

Reply:
Yep, this sounds a bit risky to me.
Can we mount it to a different directory and set path to the socket used by the supervisor as an environment variable?

Reply:
Supervisor respects `DOCKER_SOCKET` variable.

Reply:
good catch..

Reply:
I'm thinking if it would be simpler to just add a check on the existence of `/var/run/balena-engine.sock` before the `balena run` command

Reply:
And deleting it if it's a directory?

Reply:
another option would be to do:
`--mount type=bind,source=/var/run/balena-engine.sock,target=/var/run/balena-engine.sock
`

https://docs.docker.com/engine/reference/commandline/service_create/#add-bind-mounts-or-volumes

> type=bind: src is required, and specifies an absolute path to the file or directory to bind-mount (for example, src=/path/on/host/). An error is produced if the file or directory does not exist.

Reply:
+10000 on @robertgzr 's idea

Reply:
I don't think we should switch to such a wide bind mount, who knows what issues can come up with conflicts between the two systems in `/var/run`. I also think we should go for @robertgzr's idea